### ISSUE TITLE
hello,there is some wrong with the  compute_adjustment

### ISSUE THREAD

**Issue by limengyang1992** [Chumsy0725/logit-adj-pytorch#2]
![image](https://user-images.githubusercontent.com/73928007/131099098-98ca71f8-2b16-4428-af19-9ab98aadba69.png)

There is no problem with 10 classification, when 100 classification, the order is wrong
I modified it：

    label_freq = dict(sorted(label_freq.items()))
    label_freq = sorted(label_freq.items(), key=lambda e: e[1], reverse=True)
    label_freq_array = np.array([x[1] for x in label_freq])
    label_freq_array = label_freq_array / label_freq_array.sum()

**Comment by Chumsy0725** [comment]
Hey! @limengyang1992 
It can be rectified by casting the keys as integers.

>         for j in target:
>            key = int(j.item())
>            label_freq[key] = label_freq.get(key, 0) + 1

It is convenient to have adjustments sorted according to the class number than the class frequency.
Thanks!

**Comment by sadeepj** [comment]
Hi @Chumsy0725,

I didn't have a deeper look - but it looks like there's still a problem here. As seen from the screenshot, the tensor returned by the above compute_adjustment() method has the order ['0', '1', '10', '11', ...] instead of ['0', '1', '2', '3', ...]. Wouldn't this cause a problem for CIFAR-100 when this output tensor is used to correct the logits, which are always ordered ['0', '1', '2', '3', ...] ?

**Comment by Chumsy0725** [comment]
Hi Dr. @sadeepj ,

The issue of keys being out of order is corrected by casting the key as integers. The resulting `label_freq` dictionary is attached bellow.
![image](https://user-images.githubusercontent.com/58227113/132554799-d91e2c11-3779-45fb-9cad-6ba527e8f658.png)

also the resulting probability values (attached bellow) in `label_freq_array` after normalizing are the same as the values listed in [google-research/logit_adjustment](https://github.com/google-research/google-research/blob/master/logit_adjustment/data/cifar100-lt_base_probs.txt) and the values are in the same order. 
![image](https://user-images.githubusercontent.com/58227113/132555744-722d3e14-9d67-41af-b005-2121a57a2c89.png)

**Comment by sadeepj** [comment]
Thanks @Chumsy0725! I didn't realize that the issue is already fixed by https://github.com/Chumsy0725/logit-adj-pytorch/commit/2e5bd39e9aa086169e391cc77dd31fa3ba66a83c.

@limengyang1992, does this solve the issue you were facing?
